### PR TITLE
Issue #539 Add callback function for configs

### LINKED ISSUE #224: Better info message(s) after crc config set
If we set the params via `config set`, the info message printed post operation could be improved

```
 # ./crc config set memory 20480
crc - Local OpenShift 4.x cluster

# ./crc config set cpus 4      
crc - Local OpenShift 4.x cluster
```

here `crc - Local OpenShift 4.x cluster` doesn't indicate anything, either remove it or may be something like `Successfully configured memory to X`.

### LINKED ISSUE #539: Add callback function to inform user if memory/cpu config set.
As of now we are not letting users know that if they set `memory` or `cpu` configs as part of config set they need to start a fresh instance to apply those configs.

Something we use to have in minishift side.

```
$  ./minishift config set memory 4096
No Minishift instance exists. New 'memory' setting will be applied on next 'minishift start'

$ ./minishift config set memory 4096
You currently have an existing Minishift instance. Changes to the 'memory' setting are only applied when a new Minishift instance is created.
To let the configuration changes take effect, you must delete the current instance with 'minishift delete' and then start a new one with 'minishift start'.
```

### PR DESCRIPTION
**Pull request by praveenkumar**
Configs which required to provide a message to end user should
be considered using callback functions.

Fixes #539 
Fixes #224

### CODE REVIEW

File: pkg/crc/config/viper_config.go [hunk 1]
```diff
@@ -173,35 +176,47 @@ func runValidations(validations []ValidationFnType, value interface{}) (bool, st
...
 	ok, expectedValue := runValidations(allSettings[key].validationFns, value)
 	if !ok {
-		return fmt.Errorf("Config value is invalid: %s, Expected: %s\n", value, expectedValue)
+		return "", fmt.Errorf("Config value is invalid: %s, Expected: %s\n", value, expectedValue)
 	}
 
 	set(key, value)
 
-	return nil
+	callbackMsg := runCallbacks(allSettings[key].callbackFns, key, value)
```
Comment (anjannath) [inline pkg/crc/config/viper_config.go]: :+1:

File: pkg/crc/config/viper_config.go [hunk 2]
```diff
@@ -173,35 +176,47 @@ func runValidations(validations []ValidationFnType, value interface{}) (bool, st
...
+		return "", fmt.Errorf("Config property does not exist: %s", key)
 	}
 
 	if err := unset(key); err != nil {
-		return fmt.Errorf("Error unsetting config property: %s : %v", key, err)
+		return "", fmt.Errorf("Error unsetting config property: %s : %v", key, err)
 	}
 
-	return nil
+	return fmt.Sprintf("Successfully removed config for: %s", key), nil
```
Comment (anjannath) [inline pkg/crc/config/viper_config.go]: :+1:

**Comment by anjannath** [comment]
@praveenkumar Should refer to #539 in the commit message.

**Comment by praveenkumar** [comment]
> Should refer to #539 in the commit message.

Ah yes, will do that.

**Pull request merged**